test/cat.jl

```diff
@@ -88,27 +88,18 @@ module TestCat
     dt[1] = zeros(Int, nrow(dt))
     vcat(dt, alt_dt)
 
-    # Don't fail on non-matching names
-    names!(alt_dt, [:A, :B, :C])
-    vcat(dt, alt_dt)
-
     dtr = vcat(dt4, dt4)
     @test size(dtr, 1) == 8
     @test names(dt4) == names(dtr)
     @test isequal(dtr, [dt4; dt4])
 
-    dtr = vcat(dt2, dt3)
```

Comment:
The assignments are almost 100 lines prior to this, and the DataTables have differing numbers of columns and fail the new stricter vcat rules. I test for this failure somewhere else in this PR